server/src/app/user/user.service.impl.spec.ts

```diff
@@ -0,0 +1,125 @@
+import {UserServiceImpl} from './user.service.impl';
+import {anything, instance, mock, when} from 'ts-mockito';
+import {TestingModule, Test} from '@nestjs/testing';
+import {User} from '../../domain/user/user.entity';
+import {UserDto} from '../../domain/user/dto/user.dto';
+import {UserRepository} from '../../port/persistence/repository/user.repository.impl';
+
+const mockAddress = 'testAddress';
+const mockUser = new User(mockAddress);
+
+describe('UserServiceImpl', async () => {
+
+    let service: UserServiceImpl;
+    describe('dependency resolve', () => {
+        it('should be defined', async () => {
+            const mockRepoisotry: UserRepository = mock(UserRepository);
+            const mockRepositoryInst: UserRepository = instance(mockRepoisotry);
+            const module: TestingModule = await Test.createTestingModule({
+                providers: [
+                    UserServiceImpl,
+                    {
+                        provide: 'IUserRepository',
+                        useValue: mockRepositoryInst,
+                    },
+                ],
+            }).compile();
+            service = module.get<UserServiceImpl>(UserServiceImpl);
+
+            expect(service).toBeDefined();
+        });
+    });
+    describe('#get()', () => {
+        const mockUserId = 1;
+        it('should return mockUser', async () => {
+            const mockRepository: UserRepository = mock(UserRepository);
+            when(mockRepository.findById(mockUserId)).thenReturn(new Promise((resolve) => {
+                resolve(mockUser);
+            }));
+            const mockedRepositoryInst: UserRepository = instance(mockRepository);
+            service = new UserServiceImpl(mockedRepositoryInst);
+
+            expect(await service.get(mockUserId)).toBe(mockUser);
+
+        });
+
+        it('should return undefined', async () => {
+            const mockRepository: UserRepository = mock(UserRepository);
+            when(mockRepository.findById(null)).thenReturn(undefined);
+            const mockedRepositoryInst: UserRepository = instance(mockRepository);
+            service = new UserServiceImpl(mockedRepositoryInst);
+
+            expect(await service.get(null)).toBeUndefined();
+        });
+    });
+
+    describe('#create()', () => {
+
+        it('should return mockUser', async () => {
+            const mockRepository: UserRepository = mock(UserRepository);
+            when(mockRepository.findByAddress(mockAddress)).thenReturn(undefined);
+            when(mockRepository.save(anything())).thenReturn(new Promise((resolve) => {
+                resolve(mockUser);
+            }));
+            const mockedRepositoryInst: UserRepository = instance(mockRepository);
+            service = new UserServiceImpl(mockedRepositoryInst);
+            const userDto = new UserDto(mockAddress);
+
```

Comment:
i think
```
const mockRepository: UserRepository = mock(UserRepository);
when(mockRepository.findByAddress(mockAddress)).thenReturn(undefined); when(mockRepository.save(anything())).thenReturn(new Promise((resolve) => {
        resolve(mockUser);
}));
const mockedRepositoryInst: UserRepository = instance(mockRepository);
 service = new UserServiceImpl(mockedRepositoryInst);
```
duplicated above code can be moved to
```
beforeEach('set user service', async () =>{
~~~
})
```

Reply:
i don't think so. 
```
when(mockRepository.findByAddress(mockAddress)).thenReturn(undefined); when(mockRepository.save(anything())).thenReturn(new Promise((resolve) => {
        resolve(mockUser);
}));
```
is only used in '#get()' / 'should return user'.
i think it's better to define only in '#get()' / 'should return user'.